File: 0006-zigzag-conversion/0006-zigzag-conversion.swift
class Solution {
    func convert(_ s: String, _ numRows: Int) -> String {
        if numRows == 1 {return s}
        var hash = [Int:String]()
        var result = ""
        for i in 1...numRows {
            hash[i] = ""
        }

        var down = true
        var rowChecker = 1
        for c in s.map{String($0)} {
            if down {
                hash[rowChecker]! += c
                rowChecker += 1
                
                if rowChecker == numRows {
                    down = false
                }
                
            }else if !down {
                hash[rowChecker]! += c
                rowChecker -= 1

                if rowChecker == 1 {
                    down = true
                }
            }
        }

        for (i,j) in hash.sorted(by: {$0.key < $1.key}) {
            result += j
        }

        return result
    }
}

File: 0006-zigzag-conversion/0006-zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1 or numRows >= len(s):
            return s

        arr = ["" for _ in range(numRows)]
        currentRow = 0
        goingDown = False  

        for char in s:
            arr[currentRow] += char
            if currentRow == 0 or currentRow == numRows - 1:
                goingDown = not goingDown
            currentRow += 1 if goingDown else -1
        
        return ''.join(arr)
